README.md

```diff
@@ -579,9 +590,13 @@ Array
 -----
 
 Arrays are square brackets with values inside. Whitespace is ignored. Elements
-are separated by commas. Data types may not be mixed (different ways to define
-strings should be considered the same type, and so should arrays with different
-element types).
+are separated by commas.
+
+All values that are allowed in [key/value pairs](#user-content-keyvalue-pair)
```

Comment:
That's means... all values are allowed?

I'd prefer that we rephrase this paragraph as something along the lines of:

> \<prev-paragraph\> All elements within an array must have the same data type.
>
> All string values are considered the same type. All arrays (regardless of types of elements in them) are considered the same type. All inline tables (regardless of the key-value pairs in them) are considered the same type.

Reply:
@pradyunsg "All values that are allowed in key/value pairs are also allowed in arrays..." – That (partial) sentence or something similar is essential because we have to explain which types are allowed in arrays (String, Integer, Float, Boolean, Offset Date-Time, Local Date-Time, Local Date, Local Time, Array, Inline Table) and which are not (Table, Array of Tables), and currently we don't.

It might be possible to rephrase it a bit, say: "An array collects values of any of the types that are allowed in key/value pairs..." But we must be explicit about this, and a backreference to the key/value pairs it much better than having to list all (currently ten) allowed types again.

Reply:
Optionally, we could move the statements about string, array, and inline table types up to the proposed Value Types section. We'd simplify it, though, if we did that.

----

All string values are considered the same type. All arrays are considered the same type, regardless of content. All inline tables are considered the same type, regardless of content.

Reply:
Having said all that, let me propose the following synthesis of both of your suggestions. We'd also include a backreference from "value type".

---

Arrays are square brackets with values inside. Whitespace is ignored. Elements
are separated by commas.

All elements within an array must have the same value type.

_(Examples go here.)_

Reply:
> Arrays are square brackets with values inside. Whitespace is ignored. Elements
> are separated by commas.
> 
> All elements within an array must have the same value type.

LGTM.